src/test/java/seedu/address/model/task/LinkTest.java

```diff
@@ -0,0 +1,41 @@
+package seedu.address.model.task;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class LinkTest {
+    @Test
+    public void constructor_null_throwsIllegalArgumentException() {
+        assertThrows(NullPointerException.class, () -> new Link(null));
+    }
+
+    @Test
+    public void isValidLink() {
+        // null link
+        assertTrue(() -> new Link().isEmpty());
+
+        // blank link
+        assertFalse(Link.isValidLink(""));
+        assertFalse(Link.isValidLink(" "));
+
+        // missing parts
+        assertFalse(Link.isValidLink("a.com"));
+        assertFalse(Link.isValidLink("google"));
+
+        // invalid links
+        assertFalse(Link.isValidLink("www.google.com"));
+        assertFalse(Link.isValidLink("www.google.com.sg"));
```

Comment:
Regarding the link you may want to edit the error message so that user will not confuse? 🤔 Becos "www.google.com" can consider as valid URL